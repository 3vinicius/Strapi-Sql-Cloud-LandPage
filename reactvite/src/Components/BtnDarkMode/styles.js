import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    height: 80px;
    width: 50px;
    background-color: red;
    padding: 10px;
    border-radius: 25px;
    margin: 10px;
    cursor: pointer;
  `}
`;
