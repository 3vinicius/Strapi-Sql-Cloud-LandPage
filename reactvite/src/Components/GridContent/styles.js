import styled, { css } from 'styled-components';
import darkMode from '../../Utils/stylesUtils';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
    text-align: center;
    max-width: 58rem;
    margin: auto;
    font-size: ${theme.fonts.sizes.medium};
    ${darkMode(colorDark)}
    ;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge}
  `}
`;
