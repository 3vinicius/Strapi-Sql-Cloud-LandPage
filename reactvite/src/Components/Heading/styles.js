import styled, { css } from 'styled-components';

// sizes: {
//   xsmall: '0.8rem',
//   lsmall: '1.6rem',
//   medium: '2.4rem',
//   large: '3.2rem',
//   xlarge: '4.0rem',
//   xxlarge: '4.8rem',
//   huge: '5.6rem',
//   xhuge: '6.4rem',
// },
const titleSize = {
  small: (theme) => css`
  font-size: ${theme.fonts.sizes.medium} ;
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large} ;
  `,
  big: (theme) => css`
  font-size: ${theme.fonts.sizes.xhuge} ;
  ${mediaFonte(theme)}

`,
};

const mediaFonte = (theme) => css`@media ${theme.myMedias.letMedium} {
  font-size: ${theme.fonts.sizes.large};
}`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.colorWhite : theme.colors.primaryColor};
    ${titleSize[size](theme)}
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}
`;
