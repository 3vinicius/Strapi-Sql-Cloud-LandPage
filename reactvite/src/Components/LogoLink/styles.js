/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`;
  `}
  display: flex;
  align-items: center;
  > img {
    max-width: 8rem;
    max-height: 8rem;
  }
`;
