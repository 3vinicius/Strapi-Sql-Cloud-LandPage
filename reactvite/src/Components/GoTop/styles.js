import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.a`
  ${() => css``}
  position: fixed;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.colorWhite};
  display:flex;
  align-items: center;
  justify-content: center;
  height:4rem;
  width: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  transition: all 0.5s ease-out  ;
  border: 1px solid #060a18;
  box-shadow: 0px 0px 5px 2px #00000054;


  &:hover{
    opacity: 1;
  }
`;
