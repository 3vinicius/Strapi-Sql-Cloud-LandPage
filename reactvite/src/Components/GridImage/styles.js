import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
  padding: 200px
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    ;
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  overflow: hidden;
  `}
`;

export const Imagem = styled.img`
  ${() => css`
    width:100%;
    height: 100%;
    transition: all 300ms ease-in-out;

    &:hover{
      transform: scale(1.2)
    }
  `}
`;
