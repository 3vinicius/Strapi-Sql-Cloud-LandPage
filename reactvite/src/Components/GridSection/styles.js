import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    background: ${
      colorDark ? theme.colors.primaryColor : theme.colors.colorWhite
    };
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  ${Title}{
    position: relative;
    left: 5rem;
  }
  ${Title}::before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -2rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
