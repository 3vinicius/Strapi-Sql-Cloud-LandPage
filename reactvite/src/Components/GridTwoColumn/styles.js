import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as Texts } from '../TextComponent/styles';
import darkMode from '../../Utils/stylesUtils.js';

// primaryColor: string;
//         secondaryColor: string;
//         colorWhite: string;

export const Container = styled.div`
  ${({ theme, background }) => css`

    display:grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media(max-width: 850px){
      grid-template-columns: 1fr;
      text-align: center;
    }


    ${Texts}{
      ${darkMode(background)}
    }
    ${Title}{
      margin-bottom: ${theme.spacings.large};
      ${darkMode(background)}
    }

  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  @media ${theme.myMedias.letMedium}{
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImgContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Img = styled.img`
  ${({ theme }) => css`
  width: 100%;
  min-width: 200px;
  overflow: hidden;
  @media(max-width: 850px){
      width: 400px;
    }
  @media ${theme.myMedias.letMedium}{
    width:300px;
  }
  `}
`;
