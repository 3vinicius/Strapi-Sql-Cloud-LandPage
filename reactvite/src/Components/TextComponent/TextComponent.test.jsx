import { screen } from '@testing-library/react';
import { TextComponent } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<TextComponent />', () => {
  it('Should render <TextComponent />', () => {
    renderTheme(<TextComponent>Hello</TextComponent>);
    const div = screen.getByText('Hello');
    expect(div).toBeInTheDocument();
    expect(div).toMatchSnapshot();
  });

  it('Should correct text, how 24px', () => {
    renderTheme(<TextComponent>Hello</TextComponent>);
    const div = screen.getByText('Hello');
    expect(div).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
      color: theme.colors.colorWhite,
    });
  });
});
