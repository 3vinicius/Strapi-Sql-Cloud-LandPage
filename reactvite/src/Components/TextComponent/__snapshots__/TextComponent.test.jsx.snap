// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> Should render <TextComponent /> 1`] = `
.c0 {
  font-size: 2.4rem;
  color: #fff;
}

<p
  class="c0"
>
  Hello
</p>
`;
